packages/storage/amplify_storage_s3_dart/test/test_utils/mocks.dart

```diff
@@ -1,15 +1,31 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import 'dart:async';
+
 import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3_dart/amplify_storage_s3_dart.dart';
 import 'package:amplify_storage_s3_dart/src/sdk/s3.dart';
 import 'package:amplify_storage_s3_dart/src/storage_s3_service/storage_s3_service.dart';
 import 'package:amplify_storage_s3_dart/src/storage_s3_service/transfer/transfer.dart';
 import 'package:aws_signature_v4/aws_signature_v4.dart';
 import 'package:mocktail/mocktail.dart';
 import 'package:smithy/smithy.dart';
 
-class MockStorageS3Service extends Mock implements StorageS3Service {}
+class MockStorageS3Service extends Mock implements StorageS3Service {
+  @override
+  S3DownloadTask downloadData({
```

Comment:
Why does this need an override?

Reply:
I added this because I was getting an error when testing and thought adding that might help but it hasn't. Here's the error:

```
Bad state: No method stub was called from within `when()`. Was a real method called, or perhaps an extension method?
package:mocktail/src/mocktail.dart 262:7            When._completeWhen
package:mocktail/src/mocktail.dart 257:12           When.thenAnswer
test/platform_impl/download_file_io_test.dart 63:9  main.<fn>.<fn>
```